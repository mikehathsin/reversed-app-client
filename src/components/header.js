import { Fragment } from "react";
import Alert from "react-bootstrap/Alert";

export const Header = ({ error }) => {
  return (
    <Fragment>
      {error && <Alert variant="danger">Error to fetch: {error}</Alert>}
      <h1>Results:</h1>
    </Fragment>
  );
};
