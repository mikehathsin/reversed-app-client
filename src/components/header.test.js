import { render } from "@testing-library/react";

import { Header } from "./header";

describe("Header tests", () => {
  it("Should render correctly", () => {
    const component = render(<Header error={null} />);
    component.getByText("Results:");
    expect(component.container).toMatchSnapshot();
  });

  it("Should show the error message", () => {
    const error = "Network Error";
    const component = render(<Header error={error} />);
    component.getByText(`Error to fetch: ${error}`);
    expect(component.container).toMatchSnapshot();
  });
});
