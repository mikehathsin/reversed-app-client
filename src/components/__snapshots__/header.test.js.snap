// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Should render correctly 1`] = `
<div>
  <h1>
    Results:
  </h1>
</div>
`;

exports[`Header tests Should show the error message 1`] = `
<div>
  <div
    class="fade alert alert-danger show"
    role="alert"
  >
    Error to fetch: 
    Network Error
  </div>
  <h1>
    Results:
  </h1>
</div>
`;
